core/common/src/DateTimePeriod.kt

```diff
@@ -45,83 +52,267 @@ sealed class DateTimePeriod {
         if (this === other) return true
         if (other !is DateTimePeriod) return false
 
-        if (years != other.years) return false
-        if (months != other.months) return false
+        if (totalMonths != other.totalMonths) return false
         if (days != other.days) return false
-        if (hours != other.hours) return false
-        if (minutes != other.minutes) return false
-        if (seconds != other.seconds) return false
-        if (nanoseconds != other.nanoseconds) return false
+        if (totalNanoseconds != other.totalNanoseconds) return false
 
         return true
     }
 
     override fun hashCode(): Int {
-        var result = years
-        result = 31 * result + months
+        var result = totalMonths
         result = 31 * result + days
-        result = 31 * result + hours
-        result = 31 * result + minutes
-        result = 31 * result + seconds.hashCode()
-        result = 31 * result + nanoseconds.hashCode()
+        result = 31 * result + totalNanoseconds.hashCode()
         return result
     }
 
-    // TODO: parsing from iso string
+    companion object {
+        fun parse(text: String): DateTimePeriod {
```

Comment:
I believe we need `DatePeriod.parse` as well, that prevents parsing (non-zero?) components after `T`